### PR TITLE
Fixes TA Zoom URL settings

### PR DESCRIPTION
**Pull request by jacksontromero**
## Fixes the following issues:
* TAs had to enter URLs without http:// before but now **they have to**.  I added form validation so attempting to submit an invalid URL should fail and give a popup explaining that (pls test).
* On page reload, students would lose the Join Zoom button

## Changes
* There was a frontend object called helpingTAInfo that was updated by sockets.  It has now been removed and has been added as an optional property to the StudentData type (and by extension StudentDataContext)
* Helping a student and sending messages still send the TA's name for notification purposes, but the actual info of the TA is automatically updated inside StudentData
* This is why the ohq.help function now takes in so many arguments

## Dependencies
* #211 

# Announcements to be made:
* TAs who currently configured their URLs to not have the HTTP tag will have to **update their url before helping at OH again**

### CODE REVIEW

**Review by angelaz1 — COMMENTED** [review]
Only took a quick look but everything works on my end. A few suggestions & should be good to go - @amandaLi7 can take another look at this after the other PR is merged?

File: client/src/components/settings/VideoChatSettings.tsx [hunk 1]
```diff
@@ -24,15 +24,32 @@ export default function VideoChatSettings(props) {
     setVideoChatEnabled(chatEnabled);
     SettingsService.updateVideoChatSettings({
       enabled: chatEnabled,
-      url: chatEnabled ? videoChatURL : '',
+      url: videoChatURL,
     });
   };
 
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (jacksontromero) [reply client/src/components/settings/VideoChatSettings.tsx]: Should be resolved by a94949e

File: client/src/components/home/student/StudentMain.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ import {StudentDataContext} from '../../../contexts/StudentDataContext';
 
 function StudentMain() {
   const [removeConfirm, setRemoveConfirm] = useState(false);
-  const [helpingTAInfo, setHelpingTAInfo] = useState(null);
+  const [messagingTAName, setMessagingTAName] = useState(null);
```
[the remark this answers is not in the document]
Comment (jacksontromero) [reply client/src/components/home/student/StudentMain.tsx]: Should be resolved by a94949e

**Comment by pranavaddepalli** [comment]
image

**Comment by pranavaddepalli** [comment]
Future potential is to change the message so instead of "URL" it says "URL with https://" but this works for me

**Pull request merged**